src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Properties/InterceptedProjectProperties/DebugPropertyPage/RemoveStartArgumentsActionHandler.cs

```diff
@@ -0,0 +1,96 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using System.Xml;
+using Microsoft.VisualStudio.ProjectSystem.Debug;
+using Microsoft.VisualStudio.ProjectSystem.Utilities;
+using Microsoft.VisualStudio.ProjectSystem.VS.PropertyPages.Designer;
+using Microsoft.VisualStudio.Threading;
+
+namespace Microsoft.VisualStudio.ProjectSystem.Properties;
+
+[Export(typeof(ILinkActionHandler))]
+[ExportMetadata("CommandName", "RemoveStartArguments")]
+internal class RemoveStartArgumentsActionHandler : ILinkActionHandler
+{
+    private const string PropertyCommandArgsInUserFile = "StartArguments";
+    private const string UserSuffix = ".user";
+
+    private string? _xmlFile;
+    private List<string>? userFileCommandArgs;
+
+    public async Task HandleAsync(UnconfiguredProject project, IReadOnlyDictionary<string, string> editorMetadata)
+    {
+        _xmlFile = project.FullPath + UserSuffix;
+
+        await RemoveElementAsync();
+
+        await CopyUserArgsToLaunchProfileAsync(project);
+    }
+
+    private async Task RemoveElementAsync()
+    {
+        await TaskScheduler.Default;
+
+        XmlDocument doc = new() { XmlResolver = null };
+
+        try
+        {
+#pragma warning disable CA3075 // Insecure DTD processing in XML
+            doc.Load(_xmlFile);
+#pragma warning restore CA3075 // Insecure DTD processing in XML
+        }
+        catch (Exception ex)
+        {
+            TraceUtilities.TraceException("Failed to load .user file", ex);
+        }
+
+        XmlNodeList xmlNodeList = doc.GetElementsByTagName(PropertyCommandArgsInUserFile);
+
+        for (int i = xmlNodeList.Count - 1; i >= 0; --i)
+        {
+            userFileCommandArgs ??= new();
+            userFileCommandArgs.Add(xmlNodeList[i].InnerText);
+            xmlNodeList[i].ParentNode.RemoveChild(xmlNodeList[i]);
+        }
+
+        try
+        {
+            doc.Save(_xmlFile);
+        }
+        catch (Exception ex)
+        {
+            TraceUtilities.TraceException("Failed to save .user file", ex);
+        }
+
+        return;
+    }
+
+    private async Task CopyUserArgsToLaunchProfileAsync(UnconfiguredProject project)
+    {
+        ILaunchSettingsProvider? launchSettingsProvider = project.Services.ExportProvider.GetExportedValue<ILaunchSettingsProvider>();
```

Comment:
Any reason to do this rather than import the `ILaunchSettingsProvider` in a constructor?

Reply:
Among other things it would make it easier to write unit tests for this.

Reply:
I already tried importing `ILaunchSettingsProvider` in the constructor, but I got a lot of composition errors.